services/offline/src/lib/offline-provider.jsx

```diff
@@ -1,25 +1,25 @@
 import PropTypes from 'prop-types'
 import React from 'react'
-import { RecordingStatesProvider } from './cacheable-section'
-import { CachedSectionsProvider } from './cached-sections'
+import { CacheableSectionProvider } from './cacheable-section-state'
 import { OfflineInterfaceProvider } from './offline-interface'
 
 /** A context provider for all the relevant offline contexts */
-export function OfflineProvider({ offlineInterface, pwaEnabled, children }) {
+export function OfflineProvider({
+    offlineInterface,
+    cacheableSectionStore,
+    children,
+}) {
     return (
-        <OfflineInterfaceProvider
-            offlineInterface={offlineInterface}
-            pwaEnabled={pwaEnabled}
-        >
-            <RecordingStatesProvider>
-                <CachedSectionsProvider>{children}</CachedSectionsProvider>
-            </RecordingStatesProvider>
+        <OfflineInterfaceProvider offlineInterface={offlineInterface}>
+            <CacheableSectionProvider store={cacheableSectionStore}>
+                {children}
+            </CacheableSectionProvider>
         </OfflineInterfaceProvider>
     )
 }
 
 OfflineProvider.propTypes = {
+    cacheableSectionStore: PropTypes.shape({}),
```

Comment:
As above, I don't think passing in an externally-initialized store should be necessary here, what's the advantage of doing it that way?

Reply:
I think you're right - replied [above too](https://github.com/dhis2/app-runtime/pull/901#discussion_r668536789)